File: agent_misc/mcps/codeql-development-mcp-server/server/ql/java/tools/test/CallGraphFrom/Example1.java
// COMPLIANT: Methods that should not be analyzed
class Example1 {
    void unrelated1() {
        // No calls
    }

    void unrelated2() {
        unrelated1();
    }

    // NON_COMPLIANT: Source method for call graph analysis
    void sourceFunc() {
        unrelated1();
        unrelated2();
    }
}

File: agent_misc/mcps/codeql-development-mcp-server/server/ql/csharp/tools/test/CallGraphFrom/Example1.cs
// COMPLIANT: Methods that should not be analyzed
class Example1 {
    void Unrelated1() {
        // No calls
    }

    void Unrelated2() {
        Unrelated1();
    }

    // NON_COMPLIANT: Source method for call graph analysis
    void SourceFunc() {
        Unrelated1();
        Unrelated2();
    }
}
